app/models/piece.rb

```diff
@@ -1,3 +1,24 @@
 class Piece < ApplicationRecord
   belongs_to :game
+  validates :color, presence: true
+  validates :type, presence: true
+
+  scope :white, ->() { where(color: 'white') }
+  scope :black, ->() { where(color: 'black') }
+
+  def self.at(x_position, y_position)
+    Piece.find_by(x_position: x_position, y_position: y_position)
+  end
```

Comment:
Looks like this isn't used anywhere. Can be removed 😄 

Note: `Piece.find_by(x_position: x_position, y_position: y_position)` won't do what you're expecting. This will grab a piece from **any** game. Not the current game. Our database will have many games in it.